### PR TITLE
Fix #1412: add CI jobs for debug builds

### LINKED ISSUE #1412: Add CI check for debug build
**Is your feature request related to a problem? Please describe.**

Debug builds on macOS (and possibly Linux) were broken because of some shared code using Windows-only APIs.

**Describe the solution you'd like**

CI should also check that debug built fine.

**Additional context**

Example here: bf5a39c14

### PR DESCRIPTION
**Pull request by buepas**
With this PR merged, the CI also checks for macOS and linux devbuilds

### CODE REVIEW

**Comment by qaisjp** [comment]
Thank you for the pull request! I realise #1412 is a little unclear and might sound like it desires debug builds for Linux/macOS only, but I think it would be good to have it for Windows also?

**Comment by buepas** [comment]
I am not that familiar with the windows build setup. I can try to do that, but that definetly needs some more research!

**Comment by buepas** [comment]
Done

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -47,8 +47,52 @@ jobs:
...
+          DXSDK_DIR: './utils/DXFiles/'
+          CI: 'true'
+
+      - name: Create build artifacts
+        run: utils\premake5 compose_files
+
+      - uses: actions/upload-artifact@master
+        with:
+          name: InstallFiles
+          path: InstallFiles/
```
Comment (qaisjp) [inline .github/workflows/build.yaml]: Might need to set a different upload filename for the debug build? (artifacts are workflow-wide)

```suggestion

      - name: Create build artifacts
        run: utils\premake5 compose_files

      - uses: actions/upload-artifact@master
        with:
          name: InstallFiles-debug
          path: InstallFiles/
```

**Review by qaisjp — APPROVED** [review]

**Comment by buepas** [comment]
Just checked the artifacts, looks like everything works as intended.

**Comment by qaisjp** [comment]
Awesome, thank you very much for the PR!

**Pull request merged**